services/ordering/src/EShopOnAbp.OrderingService.Domain/Orders/OrderManager.cs

```diff
@@ -91,6 +91,30 @@ public async Task<Order> AcceptOrderAsync(Guid orderId, Guid paymentRequestId, s
         return await _orderRepository.UpdateAsync(order, autoSave: true);
     }
 
+    public async Task<Order> CancelOrderAsync(Guid orderId, Guid paymentRequestId, string paymentRequestStatus)
+    {
+        var order = await _orderRepository.GetAsync(orderId);
+        if (order == null)
+        {
+            throw new BusinessException(OrderingServiceErrorCodes.OrderWithIdNotFound)
+                .WithData("OrderId", orderId);
+        }
+
+        order.SetOrderCancelled(paymentRequestId, paymentRequestStatus);
+
+        // Publish order cancelled event
+        await _distributedEventBus.PublishAsync(new OrderCancelledEto
+        {
+            OrderId = order.Id,
+            OrderDate = order.OrderDate,
+            OrderNo = order.OrderNo,
+            Buyer = GetBuyerEto(order.Buyer),
+            Items = GetProductItemEtoList(order.OrderItems)
+        });
```

Comment:
This method needs to be re-designed as mentioned in specific file reviews:
`CancelOrderAsync` method should:

- Accept orderStatus argument, not payment related information. 
- It should be cancelling the order not the payment status
- Should publish event with payment request related data in order to process refunding.